tests/virt_autotest/guest_upgrade_run.pm

```diff
@@ -44,9 +44,23 @@ sub get_script_run {
     my $guest_list = get_required_var("GUEST_LIST");
 
     $pre_test_cmd = "$pre_test_cmd -p $product_upgrade -r $product_upgrade_repo -g \"$guest_list\"";
+    # Pass guest registration code used for guest installation in dynamically via GUEST_SCC_REGCODE
+    # and GUEST_SCC_REGCODE_LTSS. If they are not provided, SCC_REGCODE and SCC_REGCODE_LTSS_15 will
+    # be used. If there are multiple guest patterns, their corresponding registration codes should
+    # also be specified in the same order and separated by comma symbols. Empty value is allowed. For
+    # example, regcode_guest1,regcode_guest2,,regcode_guest4, because regcode_guest3 is emety, empty
+    # value is passed in to preserve order. For multiple guest patterns, an empty registration code
+    # for specific guest pattern will not be filled out by any default value and at the same time this
+    # means registration code is not needed for it at all.
+    my $guest_regcode = get_var("GUEST_SCC_REGCODE", "");
+    my $guest_regcode_ltss = get_var("GUEST_SCC_REGCODE_LTSS", "");
+    $guest_regcode = join(',', (get_var("SCC_REGCODE", "")) x scalar(split(',', $guest_list))) if (!$guest_regcode);
+    $guest_regcode_ltss = join(',', (get_var("SCC_REGCODE_LTSS_15", "")) x scalar(split(',', $guest_list))) if (!$guest_regcode_ltss);
     my $do_registration = check_var('GUEST_SCC_REGISTER', 'installation') ? "true" : "false";
     my $registration_server = get_var('SCC_URL', 'https://scc.suse.com');
     my $registration_code = get_var('SCC_REGCODE', 'INVALID_REGCODE');
+    $pre_test_cmd .= " -o \"$guest_regcode\"";
+    $pre_test_cmd .= " -O \"$guest_regcode_ltss\"";
     $pre_test_cmd .= " -e $do_registration";
     $pre_test_cmd .= " -s $registration_server";
     $pre_test_cmd .= " -c $registration_code";
```

Comment:
" -o \"$guest_regcode\"", this is used in autoyast during guest installation.  " -c $registration_code", is used in command line registration after guest installation. Is there any requirement in automation on the reg code of these two places? Will it be working well if they are different, or is it required to be the same?

Reply:
``` -c $registration_code``` indicates whether use registration during guest upgrade which was introduced long time ago to solve scc issue. This will not affect ``` -o "$guest_regcode``` which is used for guest installation.

Reply:
Think about this on a single guest in guest  upgrade test. It means to do registration twice, and potentially with the same or different regcode. Will it work fine for either case naturally? In which way did you validation, same or different key?

Reply:
It works fine. You can use different key or the same. Either works.